src/sentry/integrations/github/tasks/link_all_repos.py

```diff
@@ -35,52 +40,64 @@ def link_all_repos(
+    ).capture() as lifecycle:
+        integration = integration_service.get_integration(
+            integration_id=integration_id, status=ObjectStatus.ACTIVE
         )
-        return
-
-    installation = integration.get_installation(organization_id=organization_id)
-
-    client = installation.get_client()
+        if not integration:
+            # TODO: Remove this logger in favor of context manager
+            logger.error(
+                "%s.link_all_repos.integration_missing",
+                integration_key,
+                extra={"organization_id": organization_id},
+            )
+            metrics.incr("github.link_all_repos.error", tags={"type": "missing_integration"})
+            lifecycle.record_halt(str(LinkAllReposHaltReason.MISSING_INTEGRATION))
+            return
 
-    try:
-        repositories = client.get_repositories(fetch_max_pages=True)
-    except ApiError as e:
-        if installation.is_rate_limited_error(e):
+        rpc_org = organization_service.get(id=organization_id)
+        if rpc_org is None:
+            logger.error(
+                "%s.link_all_repos.organization_missing",
+                integration_key,
+                extra={"organization_id": organization_id},
+            )
+            metrics.incr(
+                f"{integration_key}.link_all_repos.error",
+                tags={"type": "missing_organization"},
+            )
+            lifecycle.record_halt(str(LinkAllReposHaltReason.MISSING_ORGANIZATION))
             return
 
-        metrics.incr(f"{integration_key}.link_all_repos.api_error")
-        raise
+        installation = integration.get_installation(organization_id=organization_id)
 
-    integration_repo_provider = get_integration_repository_provider(integration)
+        client = installation.get_client()
 
-    for repo in repositories:
         try:
-            config = get_repo_config(repo, integration_id)
-            integration_repo_provider.create_repository(repo_config=config, organization=rpc_org)
-        except KeyError:
-            continue
-        except RepoExistsError:
-            metrics.incr("sentry.integration_repo_provider.repo_exists")
-            continue
+            repositories = client.get_repositories(fetch_max_pages=True)
+        except ApiError as e:
+            if installation.is_rate_limited_error(e):
+                lifecycle.record_halt(str(LinkAllReposHaltReason.RATE_LIMITED))
+                return
+
+            metrics.incr(f"{integration_key}.link_all_repos.api_error")
+            raise
+
+        integration_repo_provider = get_integration_repository_provider(integration)
+
+        for repo in repositories:
+            try:
+                config = get_repo_config(repo, integration_id)
+                integration_repo_provider.create_repository(
+                    repo_config=config, organization=rpc_org
+                )
+            except KeyError:
+                continue
```

Comment:
do we need a halt or failure here?

Reply:
addressed offline, lets record success if we create atleast 1 repo